sky/serve/load_balancer.py

```diff
@@ -18,25 +21,135 @@
 
 logger = sky_logging.init_logger(__name__)
 
+# TODO(ejj): Major things that may be needed before merging Envoy Support.
+# This comment should go away before merging.
+#
+# - Automated Unit/Smoke Testing: Obviously this needs to be implemented before
+# merging.  For now the code was tested by hand.
+#
+# - Hitless Configuration Changes: When configured with a yaml file, Envoy must
+# be restarted to take up configuration changes.  When done naively (as the
+# PR currently does), restarting will cause all existing connections to be
+# dropped, forcing clients to retry later.  As written, this means that any
+# time a replica is added or removed, connections to all replicas will be
+# impacted.  There are two ways to deal with this:
+#   - Hot Restart:  Envoy has a mechanism for "hot restart".   When used, a new
+#   Envoy boots and begins handling connections with the new coniguration while
+#   the old one drains.
+#   - XDS:  Envoy has a robust remote update protocol that allows changes to be
+#   made to it's configuration at runtime.  XDS support may be needed anyways
+#   if we move to having multiple Envoy Load balancers that are not necessarily
+#   co-located with the controller.  As a result it may be worth skipping hot
+#   restart and building this out instead.
```

Comment:
I think this depends on how many engineering is required for the host restart. if it is only an argument (which is I expected) we should include hot restart?

Reply:
Yeah this is a bit of a rabbit hole, it sounds simple on paper, but it's definitely not zero effort.

- Will have to tweak how envoy is configured in the PR to use the host namespace (marginally less secure but it's fine).
- Will have to set up this python hot restart example script they published, seems fine, but also doesn't seem like a core part of the project, so there could be issues.
- It's not entirely obvious to me how to do this on k8s 

See this thread btw: (https://github.com/envoyproxy/envoy/issues/7187) My main takeaway is it's possible, people do it, but it's not super duper common, and it's a bit finicky. 

None of this is insurmountable, or even particularly hard.  But it's also entirely unnecessary if we need to do xDS anyways.  So that's the judgement call that needs to be made.

Reply:
I see. I thought it was only a flag before. Then directly move to xDS makes sense to me!

Reply:
latest version of the PR fixes this